backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -378,22 +379,49 @@ boolean childContains(Set<?> list, Set<?> sublist) {
 
   @Override
   public ComputeRepositoryDiffRequest.Response computeRepositoryDiff(
-      RepositoryFunction repositoryFunction, ComputeRepositoryDiffRequest request)
-      throws ModelDBException {
+      RepositoryDAO repositoryDAO, ComputeRepositoryDiffRequest request) throws ModelDBException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       session.beginTransaction();
-      RepositoryEntity repositoryEntity = repositoryFunction.apply(session);
 
-      CommitEntity internalCommitA = session.get(CommitEntity.class, request.getCommitA());
-      if (internalCommitA == null) {
+      if (!request.getBranchA().isEmpty()
+          && !request.getBranchB().isEmpty()
+          && !request.getCommitA().isEmpty()
+          && !request.getCommitB().isEmpty()) {
+        throw new ModelDBException(
+            "Branches and Commits both are not allowed in the request",
+            Status.Code.INVALID_ARGUMENT);
+      } else if ((!request.getBranchA().isEmpty() || !request.getBranchB().isEmpty())
+          && (!request.getCommitA().isEmpty() || !request.getCommitB().isEmpty())) {
         throw new ModelDBException(
-            "No such commit found : " + request.getCommitA(), Status.Code.NOT_FOUND);
+            "Branches and Commits both are not allowed in the request",
+            Status.Code.INVALID_ARGUMENT);
       }
 
-      CommitEntity internalCommitB = session.get(CommitEntity.class, request.getCommitB());
-      if (internalCommitB == null) {
-        throw new ModelDBException(
-            "No such commit found : " + request.getCommitB(), Status.Code.NOT_FOUND);
+      RepositoryEntity repositoryEntity =
+          repositoryDAO.getRepositoryById(session, request.getRepositoryId());
+
+      CommitEntity internalCommitA;
+      CommitEntity internalCommitB;
+      if (request.getBranchA().isEmpty() || request.getBranchB().isEmpty()) {
```

Comment:
I can merge commit `24325` into `master`

Reply:
extract commitA and commitB on separate ifs

Reply:
Resolved as per attached scenarios

![](https://files.slack.com/files-tmb/TEPGETLKX-F011B1ST58D-362f59c24b/screen_shot_2020-04-02_at_10.01.55_pm_720.png)